tsdb/head.go

```diff
@@ -1149,14 +1149,16 @@ func (h *Head) truncateMemory(mint int64) (err error) {
 		h.memTruncationCallBack()
 	}
 
+	// Set the minTime that new queries will see, before waiting for queries to finish.
+	h.minTime.Store(mint)
+	// Don't allow samples to be added before this time.
+	h.minValidTime.Store(mint)
+
 	// We wait for pending queries to end that overlap with this truncation.
 	if initialized {
 		h.WaitForPendingReadersInTimeRange(h.MinTime(), mint)
```

Comment:
I was just about to propose to have a reproduce/regression test for this (in addition to the comments) when I realized that now `h.MinTime()` and `mint` are equal :)

Reply:
(nice catch by the way)

Reply:
I think we could repro by adding more hooks to slow down the check and fire off the query at the right time.